rosidl_typesupport_c/src/type_support_dispatch.hpp

```
@@ -44,44 +43,49 @@ get_typesupport_handle_function(
     return handle;
   }
 
-#ifdef ROSIDL_TYPESUPPORT_C_USE_POCO
   if (handle->typesupport_identifier == rosidl_typesupport_c__typesupport_identifier) {
     const type_support_map_t * map = \
       static_cast<const type_support_map_t *>(handle->data);
     for (size_t i = 0; i < map->size; ++i) {
       if (strcmp(map->typesupport_identifier[i], identifier) != 0) {
         continue;
       }
-      Poco::SharedLibrary * lib = nullptr;
+      rcpputils::SharedLibrary * lib = nullptr;
+
       if (!map->data[i]) {
         char library_name[1024];
         snprintf(
           library_name, 1023, "%s__%s",
           map->package_name, identifier);
         std::string library_path = rcpputils::find_library_path(library_name);
         if (library_path.empty()) {
-          fprintf(stderr, "Failed to find library '%s'\n", library_name);
-          return nullptr;
+          throw std::runtime_error("Failed to find library '" + std::string(library_name) + "'");
```

Why does this `throw` instead of return `nullptr`?

Same below.